### PR TITLE
Improved apigee.ts.

### PR DESCRIPTION
**Pull request by cn007b**
Here I've moved functions from `cli.ts` into `apigee.ts` with purpose to reuse them in `main.ts`.
The only thing is changes is HTTP Client, now it relies on service account env var,
which should work for github action.

### CODE REVIEW

File: src/apigee.ts [hunk 1]
```diff
@@ -1,62 +1,166 @@
...
-		+ `?override=true`
-		+ `&serviceAccount=${encodeURIComponent(serviceAccount)}`;
-	const auth = new GoogleAuth();
+export async function deployAPIRevision(
+	client: GoogleClient,
+	orgId: string,
+	envId: string,
+	api: string,
+	revision: string,
+): Promise<{}> {
```
Comment (emarcotte) [inline src/apigee.ts:156]: Probably should document return type as Promise<ZOrgEnvironmentAPIDeployment>

File: src/apigee.ts [hunk 2]
```diff
@@ -1,62 +1,166 @@
...
+
+type OrgEnvAPIRevisionDeploymentChangeReport = z.infer<typeof ZOrgEnvAPIRevisionDeploymentChangeReport>;
+
+// https://cloud.google.com/apigee/docs/reference/apis/apigee/rest/v1/organizations.environments.apis.revisions.deployments/deploy
+const ZOrgEnvAPIRevisionDeployment = z.object({});
+
+// Apigee does not provide a nodejs/typescript client library, so lets make up a
+// schema to validate responses for them based on https://cloud.google.com/apigee/docs/reference/apis/apigee/rest
+const ZOrgEnvironmentAPIDeployments = z.object({
+	deployments: z.array(z.object({
```
[the remark this answers is not in the document]
Comment (emarcotte) [reply src/apigee.ts:38]: assuming the list and singular are the same resource, i didn't check the api.

File: src/apigee.ts (hunk 1 above)
Comment (cn007b) [reply src/apigee.ts]: fxd

File: src/apigee.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (cn007b) [reply src/apigee.ts]: fxd

File: src/apigee.ts [hunk 3]
```diff
@@ -1,62 +1,166 @@
...
+	revision: string,
+): Promise<OrgEnvAPIRevisionDeploymentChangeReport> {
+	const url = `https://apigee.googleapis.com/v1/organizations/${org}/environments/${env}/apis/${api}/revisions/${revision}/deployments:generateDeployChangeReport`
+		+ `?override=true`;
+
+	const res = await request(client, ZOrgEnvAPIRevisionDeploymentChangeReport, {
+		url,
+		method: 'POST',
+	});
+	return res;
```
[the remark this answers is not in the document]
Comment (cn007b) [reply src/apigee.ts]: done

File: src/cli.ts [hunk 4]
```diff
@@ -185,17 +42,21 @@ async function run() {
...
-		.action(async (org: string, api: string, dir: string, output: string) => {
+		.action(async (org: string, env: string, api: string, dir: string, output: string) => {
 			await prepareArchive(dir, output);
 			console.log(`Validating API ${api}`);
 			await createAPIRevision(client, org, api, output, "validate");
 			console.log("Success")
 			console.log("Creating revision of API")
 			const result = await createAPIRevision(client, org, api, output, "import");
 			console.log(`Successfully created revision ${result.revision}`);
+			console.log("Deploying API revision")
```
[the remark this answers is not in the document]
Comment (cn007b) [reply src/cli.ts:57]: now, it's configurable, in case we need it we can provide appropriate config.

**Review by emarcotte — APPROVED** [review]

**Comment by emarcotte** [comment]
we might need to bump/rebuild/push the 'dist' file for actions, it doesn't natively support typescript as i understand it

**Pull request closed without merging**